sympy/physics/quantum/qubit.py

```diff
@@ -296,15 +303,19 @@ def _eval_args(cls, args):
         # For two numbers, the second number is the number of bits
         # on which it is expressed, so IntQubit(0,5) == |00000>.
         elif len(args) == 2 and args[1] > 1:
-            need = bitcount(abs(args[0]))
-            if args[1] < need:
-                raise ValueError(
-                    'cannot represent %s with %s bits' % (args[0], args[1]))
-            qubit_values = [(args[0] >> i) & 1 for i in reversed(range(args[1]))]
-            return QubitState._eval_args(qubit_values)
+            return cls._eval_args_with_nqubits(args[0], args[1])
         else:
             return QubitState._eval_args(args)
 
+    @classmethod
+    def _eval_args_with_nqubits(cls, number, nqubits):
+        need = bitcount(abs(number))
```

Comment:
Should the type of `number` be checked somewhere? I now get
```
>>> IntQubit('11', nqubits=2)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "sympy/physics/quantum/qexpr.py", line 132, in __new__
    args = cls._eval_args(args, **old_assumptions)
  File "sympy/physics/quantum/qubit.py", line 292, in _eval_args
    return cls._eval_args_with_nqubits(args[0], nqubits)
  File "sympy/physics/quantum/qubit.py", line 312, in _eval_args_with_nqubits
    need = bitcount(abs(number))
TypeError: bad operand type for abs(): 'str'
```

Reply:
great catch!
The expectation here is IntQubit only accepts int parameters, but I agree it should be checked explicitly. The difference of constructor of Qubit and that of IntQubit would confuse users.
Let me add the type check.